spyder/plugins/projects.py

```diff
@@ -153,6 +154,9 @@ def register_plugin(self):
         self.sig_project_closed.connect(
             lambda v: self.editor.setup_open_files())
         self.recent_project_menu.aboutToShow.connect(self.setup_menu_actions)
+        self.run.connect(lambda fname: self.main.open_external_console(

```

Comment:
Please use `self.main.ipyconsole.create_cleint_for_file` instead. That's the way to run a file in a dedicated console.